src/NuGet.Core/NuGet.Protocol/RemoteRepositories/HttpFileSystemBasedFindPackageByIdResource.cs

```diff
@@ -480,13 +485,14 @@ private async Task<SortedDictionary<NuGetVersion, PackageInfo>> FindPackagesById
                         },
                         async httpSourceResult =>
                         {
-                            var result = new SortedDictionary<NuGetVersion, PackageInfo>();
+                            HashSet<NuGetVersion> result = [];
```

Comment:
>HashSet<NuGetVersion> result = []; [](http://example.com/codeflow?start=28&length=34)

maybe leave this unitialized and set it to a new dictionary in an else clause below?